File: 2015/01/d1a.py
import os

# clear terminal
def clear():
    os.system('cls' if os.name == 'nt' else 'clear')
clear()

# read input
line = open('2015/01/input_a.txt', 'r').readline().strip()

# read char by char and decide if we're moving up or down
floor = 0
for c in line:
    if c == '(':
        floor += 1
    elif c == ')':
        floor -= 1

print(floor)

File: 2015/01/d1a.cs
using System;
using System.IO;

class Program
{
    static void Main(String[] args)
    {
        FileStream f = new FileStream("2015/01/input_a.txt", FileMode.Open);
        StreamReader r = new StreamReader(f);
        
        string my_input = r.ReadLine();
        int floor = 0;

        foreach (char c in my_input)
        {
            if (c == '(')
            {
                floor++;
            } 
            else 
            {
                floor--;
            }
        }

        Console.WriteLine(floor.ToString());
    }
}